hibernate-core/src/main/java/org/hibernate/query/internal/QueryInterpretationCacheStandardImpl.java

```diff
@@ -77,14 +77,7 @@ public SelectQueryPlan resolveSelectQueryPlan(
 			Supplier<SelectQueryPlan> creator) {
 		log.tracef( "QueryPlan#getSelectQueryPlan(%s)", key );
 
-		final SelectQueryPlan cached = (SelectQueryPlan) queryPlanCache.get( key );
-		if ( cached != null ) {
-			return cached;
-		}
-
-		final SelectQueryPlan plan = creator.get();
-		queryPlanCache.put( key, plan );
-		return plan;
+		return (SelectQueryPlan) queryPlanCache.computeIfAbsent( key, k -> creator.get() );
```

Comment:
internally the following default implementation of 'computeIfAbsent' from ConcurrentMap will be run. It is a pity we cannot enjoy the atomicity in ConcurrentHashMap in JDK, but the usage of 'putIfAbsent()' is definitively desirable:
![Screen Shot 2020-06-13 at 2 46 19 PM](https://user-images.githubusercontent.com/8211458/84576716-f40cf680-ad84-11ea-9e76-4b9b7d2cde09.png)